custom/zipline/models.py

```diff
@@ -0,0 +1,153 @@
+        :param zipline_timestamp: the value for zipline_timestamp
+        :param vehicle_number: the value for vehicle_number
+        :param vehicle_id: the value for vehicle_id
+        :param additional_text: the value for additional_text
+        :param products: a list of ProductQuantity objects if this status
+        update is associated with products
+        """
+        product_json = [
+            {'code': p.code, 'quantity': p.quantity} for p in products
+        ] if products else []
+
+        return cls.objects.create(
+            order_id=order_id,
+            timestamp=datetime.utcnow(),
+            zipline_timestamp=zipline_timestamp,
+            status=status,
+            vehicle_number=vehicle_number,
+            vehicle_id=vehicle_id,
+            additional_text=additional_text,
+            products=product_json
+        )
+
+
+class EmergencyOrder(models.Model):
+    domain = models.CharField(max_length=126)
+
+    # The id of the user who initiated the order
+    requesting_user_id = models.CharField(max_length=126)
+
+    # The phone number from which the order was initiated
+    requesting_phone_number = models.CharField(max_length=126)
+
+    # The location to which the order should be delivered
+    location = models.ForeignKey('locations.SQLLocation', on_delete=models.PROTECT)
+
+    # The location code of the location, stored here redundantly so that we
+    # can always see historically what it was at the time of the request
+    location_code = models.CharField(max_length=126)
+
+    # A list of {'code': <code>, 'quantity': <quantity>} dictionaries
+    # representing the product code and quantity of the products being ordered
+    products_requested = jsonfield.JSONField(default=list)
+
+    # Same format as products_requested; represents products and quantities delivered
+    # according to the Zipline delivered status update(s)
+    products_delivered = jsonfield.JSONField(default=list)
+
+    # Same format as products_requested; represents products and quantities reported
+    # to have been received by the facility
+    products_confirmed = jsonfield.JSONField(default=list)
+
+    # The timestamp in CommCareHQ that the order was created
+    timestamp = models.DateTimeField()
+
+    # The total number of vehicles that will be used to fulfill the request
+    total_vehicles = models.IntegerField(null=True)
+
+    # The total number of attempts made while sending this emergency order to Zipline
+    zipline_request_attempts = models.IntegerField(default=0)
+
+    # One of the STATUS_* constants from EmergencyOrderStatusUpdate
+    status = models.CharField(max_length=126, default=EmergencyOrderStatusUpdate.STATUS_PENDING)
+
+    # A pointer to the EmergencyOrderStatusUpdate record representing the received status update
+    received_status = models.ForeignKey('EmergencyOrderStatusUpdate', on_delete=models.PROTECT,
+        related_name='+', null=True)
```

Comment:
I didn't see these status type specific fields used anywhere. Will these be used in a future PR, or did I miss it?